pkg/app/piped/cloudprovider/kubernetes/diff.go

```diff
@@ -75,6 +83,44 @@ func DiffList(olds, news []Manifest, opts ...diff.Option) (*DiffListResult, erro
 	return cr, nil
 }
 
+func mergeDataAndStringData(in *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	if in == nil {
+		return in, nil
+	}
+
+	gvk := in.GroupVersionKind()
+	if gvk.Kind != "Secret" {
+		return in, nil
+	}
+
+	var secret v1.Secret
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(in.Object, &secret)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(secret.StringData) == 0 {
+		return in, nil
+	}
+
+	if secret.Data == nil {
+		secret.Data = make(map[string][]byte, len(secret.StringData))
+	}
+	for k, v := range secret.StringData {
+		secret.Data[k] = []byte(v)
+	}
+
+	secret.StringData = nil
+	newObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&secret)
+	if err != nil {
+		return nil, err
+	}
+
+	unstructured.RemoveNestedField(newObj, "metadata", "creationTimestamp")
```

Comment:
Not sure why do we need to remove this field.

Reply:
Because it is implicitly added even if original secret manifest does not have.

Reply:
Thank you for your explanation.
I think we don't have to care about that adding since our diff is already dealing with that problem by this option https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/driftdetector/kubernetes/detector.go#L195.

Reply:
Thank you!
I'll delete it.